src/Symfony/Component/DependencyInjection/Attribute/AutowireCustom.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use ReflectionParameter;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+interface AutowireCustom
+{
+    public function buildReference(ContainerBuilder $container, ReflectionParameter $parameter): Reference;
```

Comment:
`public function getAutowireAttribute(): Autowire;`?

Can't the target use cases be handled by allowing one to cast one of their attribute into an instance of `Autowire`?
Attributes should be purely descriptive, so that attaching logic to them should be avoided.